src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1753,9 +1764,39 @@ def update_estimation(control_stream, model):
         s += '\n'
         tabrec = create_record(s)
         control_stream = control_stream.insert_record(tabrec)
+    elif model.dataset is not None and tables and not remove:
+        params, ops = _parse_params_from_table(str(tables[0]))
+        new_params = [col for col in cols if col not in params.rstrip().split(' ')]
+        news = params + " ".join(new_params) + " " + ops
+        new_record = create_record(news)
+        control_stream = control_stream.replace_records([tables[0]], [new_record])
+    elif model.dataset is not None and tables and remove:
+        params, ops = _parse_params_from_table(str(tables[0]))
+        if not new_pred and not new_res:
+            new_params = [
+                col for col in params.rstrip().split(' ') if col not in old_pred | old_res
+            ]
+        elif not new_pred:
+            new_params = [col for col in params.rstrip().split(' ') if col not in old_pred]
+        elif not new_res:
+            new_params = [col for col in params.rstrip().split(' ') if col not in old_res]
+        news = " ".join(new_params) + " " + ops
+        new_record = create_record(news)
+        control_stream = control_stream.replace_records([tables[0]], [new_record])
+
```

Comment:
Perhaps this code could be made clearer? To which tables are we adding and do we remove from all tables?